.github/workflows/scripts/slack_gem_notifications/notifications_methods.rb

```diff
@@ -0,0 +1,55 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+
+require 'time'
+require "httparty"
+
+def check_for_updates(watched_gems)
+  return if gem_list_empty?(watched_gems)
+  watched_gems.each do |gem_name|
+    return unless is_valid_gem?(gem_name)
+    versions = get_gem_info(gem_name)
+    send_bot(gem_name, versions) if gem_updated?(versions)
+  end
+end
+
+def gem_list_empty?(watched_gems)
+  if watched_gems.empty?
+    raise 'Nothing to see here! watched_gems cannot be empty'
+    exit
+  end
+
+  watched_gems.empty?
+end
+
+def is_valid_gem?(gem_name)
+  res = HTTParty.get("https://rubygems.org/api/v1/versions/#{gem_name}.json")
+
+  res.code == 200
+end
+
+# Gem entries are ordered by release date. The break limits versions to two versions: newest and previous.
+# TODO: If possible, specify how many gem objects we recieve from Ruby Gems API.
+def get_gem_info(gem_name)
+  gem_info = HTTParty.get("https://rubygems.org/api/v1/versions/#{gem_name}.json")
+  versions = []
+  gem_info.each do |gem|
+    versions << gem if gem['platform'] == 'ruby'
+    break if versions.size == 2
+  end
+
+  versions
+end
+
+def gem_updated?(versions)
+  Time.now.utc - Time.parse(versions[0]['created_at']) < 24 * 60 * 60
+end
+
+def send_bot(gem_name, versions)
+  newest = versions[0]['number'] if versions[0]
+  previous = versions[1]['number'] if versions[1]
+  HTTParty.post(ENV['SLACK_GEM_NOTIFICATIONS_WEBHOOK'],
```

Comment:
Hooray! You figured out a way to leverage HTTParty instead of backticking `curl`. Great work!